js/app/historyStore.js

```diff
@@ -262,7 +262,7 @@ const HistoryStore = new Lang.Class({
     // the same after a link click, factoring out this common function. When we
     // diverge in future interactions we should revisit this decomposition.
     show_ekn_id: function (ekn_id) {
-        Engine.get_default().get_object_by_id(ekn_id, null, (engine, task) => {
+        Engine.get_default().get_object_by_id(ekn_id, null, null, (engine, task) => {
```

Comment:
I guess on the more verbose side there's
get_object_by_id_for_default_app
And
get_object_by_id_for_app

But that's quite a mouthful